.github/workflows/prune_old_builds.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+set -euo pipefail
+
+KEEP=3
```

Comment:
what's the rationale for deleting all but 3? Why not keep, say, all? Is it really that expensive?

Reply:
I believe @dfcarney said the artefacts cost us money, enough that we stopped keeping artifacts elsewhere. Hence an attempt to cap the storage at something.

My intention is to have our redo stuff still able to build from source on cache miss, so it would only affect building older stuff, I think.

Reply:
Going to submit as-is, but happy to revisit if it turns out this is free or cheap.